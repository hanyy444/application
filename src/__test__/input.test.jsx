import React from "react";
import { render, screen } from '@testing-library/react';

import Input from "../components/input/input.component";

describe('Test Input component', () => {
    expect.assertions(2);
    test('renders without crashing', () => {

        const mockInput = {
            id: 1,
            type: 'email',
            placeholder: 'Email',
            name: 'email',
            role: 'email'
        }

        render(<Input {...mockInput} />)

        // Test for existance
        expect(screen.getByRole(mockInput.role)).toBeInTheDocument();
        expect(screen.getByRole(mockInput.role)).toMatchSnapshot();
    })
})
