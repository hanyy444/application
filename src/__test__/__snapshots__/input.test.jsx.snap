// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component renders without crashing 1`] = `
<input
  class="sc-dkzDqf erNWcM"
  id="1"
  name="email"
  placeholder="Email"
  role="email"
  type="email"
/>
`;
